### ISSUE TITLE
[Deployment] Parallelize build

### ISSUE THREAD

**Issue by heitorlessa** [aws-samples/aws-serverless-airline-booking#84]
`sam build` can run in parallel to improve deployment time - Currently it runs sequentially for every stack.

**Comment by heitorlessa** [comment]
SAM CLI now supports parallel builds as well as cache. This means, we could make use of CI Cache + SAM CLI build and parallel to make deployment quicker.

Two other fundamental changes to cut down deployment time until that will enable a future work of one pipeline per service are:

* [ ] Only trigger deployments for services that changed (+ shared, until they're localized)
* [ ] Convert [CI Parameter Store requests](https://github.com/aws-samples/aws-serverless-airline-booking/blob/develop/amplify.yml#L32) into a batch request. For those not being used as deployment time dependency for other stacks, for example integ/perf tests can be turned into a Map instead of multiple parameters
